assets/js/base/components/cart-checkout/order-summary/order-summary-item.js

```diff
@@ -65,28 +74,38 @@ const OrderSummaryItem = ( { cartItem } ) => {
 	} )
 		.convertPrecision( totals.currency_minor_unit )
 		.getAmount();
-	const subtotalPriceFormat = __experimentalApplyCheckoutFilter( {
-		filterName: 'subtotalPriceFormat',
-		defaultValue: '<price/>',
-		arg: {
-			lineItem: cartItem,
-		},
-		// Only accept strings.
-		validation: ( value ) =>
-			typeof value === 'string' && value.includes( '<price/>' ),
-	} );
+	const subtotalPriceFormat = useMemo(
+		() =>
+			__experimentalApplyCheckoutFilter( {
+				filterName: 'subtotalPriceFormat',
+				defaultValue: '<price/>',
+				extensions,
+				arg: {
+					lineItem: cartItem,
+				},
+				// Only accept strings.
+				validation: ( value ) =>
+					mustBeString( value ) && mustContain( value, '<price/>' ),
+			} ),
+		[ cartItem, extensions ]
+	);
 
 	// Allow extensions to filter how the price is displayed. Ie: prepending or appending some values.
-	const productPriceFormat = __experimentalApplyCheckoutFilter( {
-		filterName: 'cartItemPrice',
-		defaultValue: '<price/>',
-		arg: {
-			cartItem,
-			block: 'checkout',
-		},
-		validation: ( value ) =>
-			typeof value === 'string' && value.includes( '<price/>' ),
-	} );
+	const productPriceFormat = useMemo(
+		() =>
+			__experimentalApplyCheckoutFilter( {
+				filterName: 'cartItemPrice',
+				defaultValue: '<price/>',
+				extensions,
+				arg: {
+					lineItem: cartItem,
+					context: 'checkout',
```

Comment:
Is it intentional to differentiate between `context: 'summary'` and `context: 'checkout'`? Not blocking but I think it might be good to document which contexts will exist so it's easier for us (and 3rd-parties) to know which context to use and also to avoid introducing new ones by mistake.

Another idea would be to save them in constants exported from `@woocommerce/blocks-checkout`, that would help to avoid typos.

Reply:
> Is it intentional to differentiate between context: 'summary' and context: 'checkout'?

No, just something I forget to update, checkout is for the main checkout area, the summary is for the sidebar (applies for both cart and checkout sidebars).

> Another idea would be to save them in constants exported from @woocommerce/blocks-checkout, that would help to avoid typos.

I agree but I'm afraid it might complicate things beyond how complicated they're.

I know they're optional and people can just use a string, but once you implement it, you will also have it in a tutorial, and that would be an extra struggle.

Do you think the benefit of having a constant here is worth the extra overhead?

Reply:
> I think it's good to provide the context, that's why in the first version of our filters (before we had this API) I wanted to pass the block type. It doesn't cost us anything to send it, and it has the potential to be extra helpful to anything extending blocks to know the context.

I think context is helpful for filters or items that are used in several places (cart item being used in several places) but would it make sense for things used just once in a single place? feels a bit redundant.

I get the argument that it doesn't cost much to have, but afraid it might cause confusion.
It should be mandatory once you introduce the same filter somewhere else.

Reply:
> Do you think the benefit of having a constant here is worth the extra overhead?

A couple of benefits I see on using constants is making it more difficult to make typos and that if some day we want to merge two contexts or rename one of them, we will be able to do so because extensions will not be using the context name directly but a constant that we can remap to a different value. But I don't have a strong opinion on this, so I'm fine either way.